src/metabase/notification/core.clj

```diff
@@ -37,8 +40,15 @@
               :template
               :recipients))
 
-(mu/defn send-notification!
-  "Send the notification to all handlers synchronously."
+(defonce ^:private pool
+  (Executors/newFixedThreadPool
+   10 ;; TODO: this should be configurable
+   (.build
+    (doto (BasicThreadFactory$Builder.)
+      (.namingPattern "send-notification-thread-pool-%d")))))
```

Comment:
Why not just put this in an internal defsetting to start?

You'd need to put this definition in a delay then, so the database is ready in case the setting has been saved.

Reply:
good idea 5d350a27df